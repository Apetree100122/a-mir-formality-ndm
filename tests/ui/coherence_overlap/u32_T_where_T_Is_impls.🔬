// Test that we detect "indirect" overlap -- here `Foo` is implemented for `u32`
// and also all `T: Is`, and `u32: Is`.
[
    crate core {
        trait Foo<> where [] {}
        impl<> Foo<> for u32 where [] {}
        impl<ty T> Foo<> for T where [T: Is<>] {}

        trait Is<> where [] {}
        impl<> Is<> for u32 where [] {}
    }
]
