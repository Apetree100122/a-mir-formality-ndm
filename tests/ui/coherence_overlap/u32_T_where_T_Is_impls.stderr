Error: impls may overlap:
impl <> Foo < > for (rigid (scalar u32)) where [] { }
impl <ty> Foo < > for ^ty0_0 where [^ty0_0 : Is < >] { }
